File: documentation-examples-java/src/java/dev/typr/foundations/docs/routines/OracleExample.java
package dev.typr.foundations.docs.routines;

import dev.typr.foundations.*;

import java.math.BigDecimal;
import java.sql.SQLException;

@SuppressWarnings("unused")
public class OracleExample {
    Transactor tx = null; // placeholder

    //start
    // Works with any database — just use the right types
    void example() {
        var applyDiscount =
            DbProcedure.define("apply_discount")
                .input(OracleTypes.number)
                .inout(OracleTypes.varchar2)
                .build();

        var getBalance =
            DbFunction.define(
                    "get_balance", OracleTypes.number)
                .input(OracleTypes.varchar2)
                .build();
    }
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/routines/OracleExample.scala
package dev.typr.foundations.docs.routines
import dev.typr.foundationssc.*

@SuppressWarnings(Array("unused"))
object OracleExample:
  //start
  // Works with any database — just use the right types
  val applyDiscount =
    DbProcedure.define("apply_discount")
      .in(OracleTypes.number)         // amount IN
      .inout(OracleTypes.varchar2)    // status INOUT
      .build()

  val getBalance =
    DbFunction.define("get_balance", OracleTypes.number)
      .in(OracleTypes.varchar2)       // account_id
      .build()
  //stop
